src/cargo/util/errors.rs

```diff
@@ -286,6 +286,20 @@ impl From<Box<CargoError>> for CliError {
 }
 
 // =============================================================================
+// NetworkError trait
+
+pub trait NetworkError: CargoError {
+    fn maybe_spurious(&self) -> bool;
+}
+
+impl NetworkError for git2::Error {
+    fn maybe_spurious(&self) -> bool { true }
+}
+impl NetworkError for curl::ErrCode {
+    fn maybe_spurious(&self) -> bool { true }
```

Comment:
Im having a heck of a time understand how to use the curl errors.  I believe this should work. 

```curl::ErrCode::COULDNT_CONNECT```

But believing it doesn't make it work. Thoughts on how I can use the names? Should I give up and use the numbers instead?